src/vaults/SyncRequests.sol

```diff
@@ -197,6 +199,36 @@ contract SyncRequests is BaseInvestmentManager, ISyncRequests {
         }
     }
 
+    function setValuation(uint64 poolId, bytes16 scId, address asset, uint256 tokenId, address valuation_) external {
+        valuation[poolId][scId][asset][tokenId] = IERC7726(valuation_);
+
+        emit SetValuation(poolId, scId, asset, tokenId, address(valuation_));
+    }
+
+    function priceAssetPerShare(uint64 poolId, bytes16 scId, uint128 assetId)
+        public
+        view
+        returns (D18 price, uint64 computedAt)
+    {
+        (address asset, uint256 tokenId) = poolManager.idToAsset(assetId);
+        IERC7726 valuation_ = valuation[poolId][scId][asset][tokenId];
+
+        if (address(valuation_) == address(0)) {
+            return poolManager.priceAssetPerShare(poolId, scId, assetId, false);
+        } else {
+            address shareToken = poolManager.shareToken(poolId, scId);
+            (uint8 assetDecimals,) = PriceConversionLib.getPoolDecimals(shareToken, asset, tokenId);
+
+            uint128 assetAmountPerShare = valuation_.getQuote(assetDecimals, asset, shareToken).toUint128();
```

Comment:
Shouldn't it be `valuation_.getQuote(10**assetDecimals, asset, shareToken)`?

and likewise on the line below

Reply:
Yes of course, sorry!

Reply:
[a804726](https://github.com/centrifuge/protocol-v3/pull/184/commits/a80472633eb09934b7b09afd8b252a50e909af25)